### ISSUE TITLE
API Proposal - CollapsibleContainer

### ISSUE THREAD

**Issue by saroad2** [beeware/toga#1174]
**Is your feature request related to a problem? Please describe.**
It occurred to me that right now *Toga* doesn't have a collapsible widget (aka. a "box" that can be expanded or collapsed by usually clicking a button with an arrow at the top of it).

I was about to implement this new widget in Windows when I figured out that while WPF has the [expander](https://docs.microsoft.com/en-us/dotnet/api/system.windows.controls.expander?redirectedfrom=MSDN&view=net-5.0) class, Windows form does not. I tried to find a similar widget for Mac but couldn't find either.

It seems too basic not to be supported by all of these platforms and consequentially in *Toga*. Am I missing something?

ATM, we could implement this feature by creating a `toga.Box` with a `toga.Button` that toggles the visibility of this box by pressing on it. However, by doing so we'll lose the "sweeping" effect most collapsible I've seen had.

What are your thoughts?

**Comment by SamSchott** [comment]
A great idea!

This does exist in macOS and is called "disclosure control". It is described in the [HIG](https://developer.apple.com/design/human-interface-guidelines/macos/buttons/disclosure-controls/). Cocoa provides a button style `NSDisclosureBezelStyle` but actually making any widgets appear / disappear on click must be done manually AFAIK.

**Comment by freakboy3742** [comment]
I agree that this is a sufficiently common visual element that it warrants having a widget of it's own. A really simple version can be built out of a Box/Button pair as you describe; animations would be a nice-to-have as well.
